File: Day1-Historian Hysteria/question2.go
package main

import (
	"fmt"
	"math"
)

func calculateSecondQuestionSum(left, right []int) int64 {
	secondSum := int64(0)
	for _, value := range left {
		count := 0
		for _, v := range right {
			if v == value {
				count++
			}
		}
		secondSum += int64(math.Abs(float64(value * count)))
	}
	fmt.Println("secondSum", secondSum)
	return secondSum
}

File: Day1-Historian Hysteria/question2.js
// import { left, right } from "./soln";
const { left, right } = require("./soln");
function calculateSumOfRepeatedLeftHandNumbers(leftCol, rightCol) {
  let sum = 0;
  for (num of leftCol) {
    let count = 0;
    for (let i = 0; i < rightCol.length; i++) {
      if (rightCol[i] === num) {
        count++;
      }
    }
    sum += num * count;
  }
  return sum;
}

console.log(calculateSumOfRepeatedLeftHandNumbers(left, right));
